### ISSUE TITLE
sii.c parse_content infinite loop

### ISSUE THREAD

**Issue by safarir** [synapticon/siitool#11]
When the SII parser (sii.c:707) hit a SII_CAT_NOP section, the parser enter a infinite loop.

I don't have access to the eeprom layout documentation so I can't confirm the normal behavior, but I feel the buffer variable should be increased by some value.

I discovered the problem while trying to parse the default eeprom value of a AB&T EasyCAT PRO board. I can include a copy of the eeprom value for testing if necessary.

**Comment by fjes** [comment]
Thanks for reporting. It would be very nice if you could provide the EEPROM file of the board.

The EEPROM files end with 0xffff so even if some sections are unknown to the siitool at the end it should abort. But let me look into this a little bit deeper.

**Comment by safarir** [comment]
The current code will never reach the 0xffff (if it was present) because the buffer value is never incremented in the SII_CAT_NOP case.

Also, if the eeprom is corrupted/malformed and there is no 0xffff, you would end up with a out of bound read condition. There should probably be a bound check at the begining of the while loop 

```
if (buffer >= eeprom+maxsize) {
                        // Maybe report error here
			break;
}
```

It look like my eeprom was not properly programmed and is mostly empty:
[easycatpro.bin.gz](https://github.com/synapticon/siitool/files/5076382/easycatpro.bin.gz)

**Comment by fjes** [comment]
Thanks again for reporting I fixed it with 24459a2
